GVFS/GVFS.Common/NuGetUpgrade/NuGetUpgrader.cs

```diff
@@ -492,6 +522,66 @@ private bool TryRecursivelyDeleteInstallerDirectory(out string error)
             return true;
         }
 
+        private IList<IPackageSearchMetadata> QueryFeedWithRetry()
+        {
+            int retryCount = 0;
+            IList<IPackageSearchMetadata> queryResults = null;
+
+            do
+            {
+                try
+                {
+                    queryResults = this.nuGetFeed.QueryFeedAsync(this.nuGetUpgraderConfig.PackageFeedName).GetAwaiter().GetResult();
+                    return queryResults;
+                }
+                catch (Exception ex) when (ex is FatalProtocolException &&
+                                           ex.InnerException != null &&
+                                           ex.InnerException is System.Net.Http.HttpRequestException &&
+                                           ex.InnerException.Message.Contains("401 (Unauthorized)") &&

```

Comment:
Would there be any downsides to just checking for "401"?  I wasn't sure what the chances are that the exception message could change.

Reply:
Additionally, should we check for 403 in case we have a PAT scope issue that could be resolved by reacquiring?

Reply:
Sounds good - I will check just for the status code, and will check for 403 as well (and check if there are other status codes to retry on as well).